File: java-spring-webflux/src/main/java/com/zim32/benchmark/model/Triangle.java
package com.zim32.benchmark.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Triangle implements FigureInterface{
    private final int sizeA;
    private final int sizeB;
    private final int sizeC;

    @JsonCreator
    Triangle(@JsonProperty("sizeA") int sizeA, @JsonProperty("sizeB") int sizeB, @JsonProperty("sizeC") int sizeC) {
        this.sizeA = sizeA;
        this.sizeB = sizeB;
        this.sizeC = sizeC;
    }

    @Override
    public String getType() {
        return "triangle";
    }

    @Override
    public double computeArea() {
        double a = sizeA;
        double b = sizeB;
        double c = sizeC;
        var s = (a + b + c) / 2;

        return Math.sqrt(s * (s - a) * (s - b) * (s - c));
    }
}

File: kotlin-ktor-jvm/src/main/kotlin/com/example/model/Triangle.kt
package com.example.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.math.sqrt

@Serializable
@SerialName("triangle")
class Triangle(
    private val sizeA: Int,
    private val sizeB: Int,
    private val sizeC: Int,
): Figure {
    override fun getType(): String {
        return "triangle";
    }

    override fun computeArea(): Double {
        val a = sizeA.toDouble()
        val b = sizeB.toDouble()
        val c = sizeC.toDouble()

        val s = (a + b + c) / 2;

        return sqrt(s * (s - a) * (s - b) * (s - c))
    }
}